agents/src/voice/agent_activity.ts

```
@@ -313,18 +348,52 @@ export class AgentActivity implements RecognitionHooks {
     }
     // TODO(shubhra): handle tool calls
 
-    const message = ChatMessage.create({
-      role: ChatRole.ASSISTANT,
-      text: textOutput.text,
-    });
-    chatCtx.insertItem(message);
-    this.agent._chatCtx.insertItem(message);
-    this.agentSession._conversationItemAdded(message);
+    await speechHandle.waitIfNotInterrupted(tasks.map((task) => task.result));
 
-    await Promise.all(tasks);
+    // TODO(shubhra): add waiting for audio playout in audio output
 
-    this.logger.info({ speech_id: speechHandle.id }, 'playout completed');
-    speechHandle.markPlayoutDone();
+    if (speechHandle.interrupted) {
+      this.logger.debug(
+        { speech_id: speechHandle.id },
+        'Aborting all pipeline reply tasks due to interruption',
+      );
+      replyAbortController.abort();
+      await Promise.allSettled(tasks.map((task) => task.cancelAndWait));
```

btw `cancelAndWait`, when take no timeout parameter, will wait indefinitely. Do we want to pass a default timeout threshold (5s?).

---

We wait indefinitely in python as well - but yeah I think it's good to have an explicit timeout in the resource clean up. I'll add one.

---

If any of the tasks fails to settle, the next turn won't take place (agent can't speak since main loop in `AgentAcitivty` will be blocked as `speechHandle.markPlayoutDone()` will never be reached. 

I still think it's better to throw a timeout error instead of just handing so I'll add one.